lib/osf-components/addon/components/editable-field/license-manager/component.ts

```diff
@@ -92,21 +108,48 @@ export default class LicenseManagerComponent extends Component implements Licens
     @action
     changeLicense(selected: License) {
         this.set('selectedLicense', selected);
-        this.node.setNodeLicenseDefaults(selected.requiredFields);
+        this.setNodeLicenseDefaults(selected.requiredFields);
+    }
+
+    setNodeLicenseDefaults(requiredFields: Array<keyof NodeLicense>): void {
+        const {
+            copyrightHolders = '',
+            year = new Date().getUTCFullYear().toString(),
+        } = (this.changeset.get('nodeLicense') || {});
+
+        const nodeLicenseDefaults: NodeLicense = {
+            copyrightHolders,
+            year,
+        };
+
+        // Only set the required fields on nodeLicense
+        const props = requiredFields.reduce(
+            (acc, val) => ({ ...acc, [val]: nodeLicenseDefaults[val] }),
+            {},
+        );
+        set(this.changeset, 'nodeLicense', props);
     }
 
     @action
-    onSave() {
+    async save() {
+        try {
+            await this.changeset.save({});
+            this.node.save();
+            this.setProperties({
+                currentLicense: this.selectedLicense,
+                currentNodeLicense: { ...this.node.nodeLicense },
+            });
+            this.set('requestedEditMode', false);
+        } catch (e) {
+            this.toast.error(this.intl.t('registries.registration_metadata.edit_license.error'));
+        }
         this.toast.success(this.intl.t('registries.registration_metadata.edit_license.success'));
-        this.setProperties({
-            currentLicense: this.selectedLicense,
-            currentNodeLicense: { ...this.node.nodeLicense },
-        });
-        this.set('requestedEditMode', false);
     }
 
     @action
-    onError() {
-        this.toast.error(this.intl.t('registries.registration_metadata.edit_license.error'));
+    updateNodeLicense(key: any, newValue: any) {
```

Comment:
let's avoid `any` types to the extent we can -- i think `string` would be fine, if there's not a more specific type for `key`
```suggestion
    updateNodeLicense(key: string, newValue: string) {
```